test/intl402/RelativeTimeFormat/constructor/constructor/options-toobject-prototype.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.RelativeTimeFormat
+description: Checks handling of non-object option arguments to the RelativeTimeFormat constructor.
+info: |
+    InitializeRelativeTimeFormat (relativeTimeFormat, locales, options)
+features: [Intl.RelativeTimeFormat]
+---*/
+
+Object.defineProperties(Object.prototype, {
+  "style": {
+    value: "short",
+  },
+  "numeric": {
+    value: "auto",
+  },
+})
+
+const optionsArguments = [
+  true,
+  "test",
+  7,
+  Symbol(),
+];
+
+for (const options of optionsArguments) {
+  const rtf = new Intl.RelativeTimeFormat([], options);
+  const resolvedOptions = rtf.resolvedOptions();
+  assert.sameValue(resolvedOptions.style, "short",
```

Comment:
```
1.1.1 InitializeRelativeTimeFormat ( relativeTimeFormat, locales, options )

...
12. Let s be ? GetOption(options, "style", "string", «"long", "short", "narrow"», "long").
13. Set relativeTimeFormat.[[Style]] to s.
14. Let numeric be ? GetOption(options, "numeric", "string", «"always", "auto"», "always").
15. Set relativeTimeFormat.[[Numeric]] to numeric.
```

```
1.4.5 Intl.RelativeTimeFormat.prototype.resolvedOptions ()

...
5. For each row of Table 1, except the header row, do
  a. Let p be the Property value of the current row.
  b. Let v be the value of pr's internal slot whose name is the Internal Slot value of the current row.
  c. Assert: v is not undefined.
  d. Perform ! CreateDataPropertyOrThrow(options, p, v).
```

The proposal spec gives some ideas that we could give more strength to the tests:

- Considering [ToObject](https://tc39.github.io/ecma262/#sec-toobject), set the properties directly on the respective constructors prototype, not Object.prototype.
- Use accessor descriptors to store when the value is capture - when the new instance is created - and check again that `resolvedOptions` won't verify the values again. ~~An alternative - but weaker - would be changing the options values before calling resolvedOptions~~.
- Verify the descriptor of DataProperty, not only the Get operation.